TheMovieDb/Home/Contoller/HomeViewController.swift

```diff
@@ -7,21 +7,226 @@
+        case .popular:
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: HightSectionCell.reuseIdentifier, for: indexPath) as? HightSectionCell else {
+                return HightSectionCell()
+            }
+            var viewModel = MovieViewModel(movie: movie)
+            viewModel.isHighSection = true
+            cell.viewModel = viewModel
+            return cell
+        case .topRated:
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: TopRatedSectionCell.reuseIdentifier, for: indexPath) as? TopRatedSectionCell else {
+                return TopRatedSectionCell()
+            }
+          
+            var viewModel = MovieViewModel(movie: movie)
+            viewModel.numerTop = indexPath.row
+            cell.viewModel = viewModel
+            return cell
+            
+        case .upcoming, .trending, .playingNow:
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: DefaultSectionCell.reuseIdentifier, for: indexPath) as? DefaultSectionCell else {
+                return DefaultSectionCell()
+            }
+            let viewModel = MovieViewModel(movie: movie)
+            cell.viewModel = viewModel
+            return cell
+        }
+        
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        
+        guard let header = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: HomeHeader.reuseIdentifier, for: indexPath) as? HomeHeader else {
+            return HomeHeader()
+        }
+        
+        let section = MovieGroupSections(rawValue: indexPath.section)
+        if section != .popular {
+            header.nameHeader = section
+        }
+        return header
+    }
+}
+
+// MARK: - UICollectionViewControllerDelegate
+extension HomeViewController {
+    
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let section = MovieGroupSections(rawValue: indexPath.section) ?? .topRated
+        guard let movies = movies[section] else { return }
+        let movie = movies[indexPath.row]
+        let controller = DetailViewController(with: movie)
+        navigationController?.pushViewController(controller, animated: true)
+        
+    }
+    
+}
+
+extension HomeViewController {
+     func configureCollectionViewLayout() -> UICollectionViewCompositionalLayout {
+        let sectionProvider = { (sectionIndex: Int, _: NSCollectionLayoutEnvironment) -> NSCollectionLayoutSection? in
+            
+            var section: NSCollectionLayoutSection?
+            
+            switch sectionIndex {
+            case 0:
+                section = HomeViewController.getHightLayoutSection()
+            case 3:
+                section = HomeViewController.getTopRatedLayoutSection()
```

Comment:
use a enum instead since this could be interpreted as a magical numbers

Reply:
magical number disappeared